exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -172,22 +214,28 @@ func (prwe *prwExporter) PushMetrics(ctx context.Context, md pmetric.Metrics) er
 	case <-prwe.closeChan:
 		return errors.New("shutdown has been called")
 	default:
-
-		tsMap, err := prometheusremotewrite.FromMetrics(md, prwe.exporterSettings)
-		if err != nil {
-			prwe.telemetry.recordTranslationFailure(ctx)
-			prwe.settings.Logger.Debug("failed to translate metrics, exporting remaining metrics", zap.Error(err), zap.Int("translated", len(tsMap)))
-		}
-
-		prwe.telemetry.recordTranslatedTimeSeries(ctx, len(tsMap))
-
-		var m []*prompb.MetricMetadata
-		if prwe.exporterSettings.SendMetadata {
-			m = prometheusremotewrite.OtelMetricsToMetadata(md, prwe.exporterSettings.AddMetricSuffixes)
+		if prwe.enableSendingRW2 {
```

Comment:
This looks incorrect 🤔 

If `enableSendingRW2` is false, then we never call `pushMetricsV1`

Reply:
Ah ok, I see the else few lines below. Hmmm, do we really need this boolean? Maybe we could get away only with the field that chooses the PRW version and a big warning that v2 isn't working yet?

Reply:
Part of it may be the indentation.  For feature gates, I usually like to structure code like:

```golang
if !enableSendingRW2FeatureGate.IsEnabled() {
    // legacy behavior if the feature gate is disabled
    return prwe.pushMetricsV1(ctx, md)
}
// new behavior here
```

Because it makes it easy to remove the feature gate in the future.

Reply:
applied this :).